### ISSUE TITLE
Incorrect import statement detection

### ISSUE THREAD

**Issue by connorlanigan** [skuzzle/restrict-imports-enforcer-rule#2]
Using a variable called `imported` produces a `StringIndexOutOfBoundsException`, as `de.skuzzle.enforcer.restrictimports.impl.ImportMatcherImpl#isImport` wrongly detects this line as an import statement. The search for the expectedly following ` ` (space) leads to the aforementioned exception.

(The method `de.skuzzle.enforcer.restrictimports.impl.ImportMatcherImpl#isPackage` will likely be prone to the same issue.)

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce (check-imports) on project swp: Execution check-imports of goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce failed: String index out of range: -1
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution check-imports of goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce failed: String index out of range: -1
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1960)
	at de.skuzzle.enforcer.restrictimports.impl.ImportMatcherImpl.extractPackageName(ImportMatcherImpl.java:94)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:373)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:373)
	at java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:373)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at de.skuzzle.enforcer.restrictimports.impl.ImportMatcherImpl.matchFile(ImportMatcherImpl.java:47)
	at de.skuzzle.enforcer.restrictimports.impl.SourceTreeAnalyzerImpl.lambda$analyze$2(SourceTreeAnalyzerImpl.java:27)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:373)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at de.skuzzle.enforcer.restrictimports.impl.SourceTreeAnalyzerImpl.analyze(SourceTreeAnalyzerImpl.java:28)
	at de.skuzzle.enforcer.restrictimports.RestrictImports.execute(RestrictImports.java:46)
	at org.apache.maven.plugins.enforcer.EnforceMojo.execute(EnforceMojo.java:193)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	... 20 more
```

**Comment by skuzzle** [comment]
Can you supply a minimal sample source file to reproduce?

**Comment by connorlanigan** [comment]
_DataImporter.java_:
```
package example;

public class DataImporter {

	public int importData(String[] inputData){
	 	int importedLines;
		for(String line : inputData){
			doImport(line);
	 		importedLines++;
		}
		return importedLines;
	}

}
```

_pom.xml_:
```
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns="http://maven.apache.org/POM/4.0.0"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
	<modelVersion>4.0.0</modelVersion>

	<groupId>example</groupId>
	<artifactId>example</artifactId>
	<version>0.0.1</version>

	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-enforcer-plugin</artifactId>
				<version>1.4.1</version>
				<dependencies>
					<dependency>
						<groupId>de.skuzzle.enforcer</groupId>
						<artifactId>restrict-imports-enforcer-rule</artifactId>
						<version>0.4.0</version>
					</dependency>
				</dependencies>
				<configuration>
					<rules>
						<restrictImports
							implementation="de.skuzzle.enforcer.restrictimports.RestrictImports">
							<basePackage>**</basePackage>
							<bannedImports>
								<bannedImport>java.util.ArrayList</bannedImport>
							</bannedImports>
						</restrictImports>
					</rules>
				</configuration>
				<executions>
					<execution>
						<id>check-imports</id>
						<phase>process-sources</phase>
						<goals>
							<goal>enforce</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>
</project>
```

**Comment by skuzzle** [comment]
Workaround in https://github.com/skuzzle/restrict-imports-enforcer-rule/commit/67fc99cad1f0731ab7853ebd6a3b43e656faafb0 (accidently referenced wrong issue -.-)

**Comment by skuzzle** [comment]
Released in 0.5.0
